NitroxServer/ConsoleCommands/ChangeServerGamemodeCommand.cs

```diff
@@ -10,24 +11,33 @@ namespace NitroxServer.ConsoleCommands
 {
     internal class ChangeServerGamemodeCommand : Command
     {
-        private readonly ServerConfig serverConfig;
         private readonly PlayerManager playerManager;
+        private readonly ServerConfig serverConfig;
 
-        public ChangeServerGamemodeCommand(ServerConfig serverConfig, PlayerManager playerManager) : base("changeservergamemode", Perms.ADMIN, "Changes server gamemode")
+        public ChangeServerGamemodeCommand(PlayerManager playerManager, ServerConfig serverConfig) : base("changeservergamemode", Perms.ADMIN, "Changes server gamemode")
         {
-            this.serverConfig = serverConfig;
-            this.playerManager = playerManager;
             AddParameter(new TypeEnum<ServerGameMode>("gamemode", true));
+
+            this.playerManager = playerManager;
+            this.serverConfig = serverConfig;
         }
 
         protected override void Execute(CallArgs args)
         {
             ServerGameMode sgm = args.Get<ServerGameMode>(0);
 
-            serverConfig.GameMode = sgm;
-            playerManager.SendPacketToAllPlayers(new GameModeChanged(sgm));
+            if (serverConfig.GameMode != sgm)
+            {
+                serverConfig.GameMode = sgm;
+                NitroxConfig.Serialize(serverConfig);
```

Comment:
This causes existing changes to config file to be overwritten by whatever the settings were in memory. It should deserialize, make the changes, and then serialize again (maybe turn this into a method).